add Comment component

sowasred2012 opened the pull request.

adrianclay committed:
Add rank component

camillavk committed:
Add points component

Commit:
add Comment component

Commit:
add Comment component

Commit:
rename Comment component to CommentCount

Commit:
rename Comment component to CommentCount

camillavk commented on components/CommentCount.js:
```diff
@@ -0,0 +1,21 @@
+
+class CommentCount extends React.Component {
+  formatString (count) {
+    if (count == undefined) { count = 0 }
```
If commentCount is a required prop in the NewsItem, will `count` ever be undefined?

sowasred2012 replied:
My answer ties into both these comments - I'm not sure, cos with the stories it feels like a "not upvoted" story doesn't care whether it's got comments or not. Maybe the stories need to be more all-encompassing? In any case, that's the situation when `commentCount` would be undefined (I ran into it).

adrianclay replied:
Upvoting and comment count feel separate to me.

We should always be able to count the comments, no comments = 0?

rrpff replied:
You should define `propTypes` for this component, and add the `count` prop. Since you're worried about `count` being undefined, maybe you could define `defaultProps` and set it to 0 by default?

camillavk commented on stories/index.js:
```diff
@@ -22,6 +23,9 @@ storiesOf('NewsItem', module)
     <NewsItem upvoted={false} />
   ))
+  .add('with 12 comments', () => (
+    <NewsItem commentCount={12} />
```
Does the NewsItem have to have an `upvoted` prop? I think it was set as required in the `NewsItem.propTypes`?

adrianclay commented on components/NewsItem.js:
```diff
@@ -1,10 +1,14 @@
   constructor (props, context) {
     super(props, context)
-    this.state = {upvoted: this.props.upvoted}
+    this.state = {
```
I'm feeling very _conflicted_

HAHAHA it was a git joke.

adrianclay replied:
![image](https://cloud.githubusercontent.com/assets/976274/26365099/7b4a5952-3fde-11e7-949d-8cd2c4a43e10.png)

camillavk replied:
hahahaha

rrpff commented on components/CommentCount.js:
```diff
@@ -0,0 +1,21 @@
+    if (count == undefined) { count = 0 }
+
+    if (count == 1) {
+      return (count + " comment")
```
You could use template strings with interpolation to make this a bit neater:

```js
return `${count} comment`
```

rrpff commented on components/CommentCount.js:
```diff
@@ -0,0 +1,21 @@
+  formatString (count) {
+    if (count == undefined) { count = 0 }
+
+    if (count == 1) {
```
Triple equality (`===`) is preferred in JS, it won't try and coerce the values.

```js
1 == '1' // true
1 === '1' // false
```

camillavk committed:
Rename to PointsCount, remove  and only increase on upvote

camillavk committed:
Remove duplicate setState of pointsCount

adrianclay committed:
Reformat with 2 spaces

adrianclay committed:
Take .editorconfig from facebook/react

  https://github.com/facebook/react/blob/master/.editorconfig

rrpff committed:
Dedicate the whole of Week 2 to testing

- Add additional testing criteria to Week 2.
- Move context criteria to Week 3.

rrpff committed:
Week 1 Results

rrpff committed:
Setup Jest

rrpff committed:
Add NewsItemList component

rrpff committed:
Continue testing in Week 3

rrpff committed:
Move components into directories

rrpff committed:
Split stories into a file for each component

rrpff committed:
Add session and homework sections to each week

adrianclay committed:
Add the prop-types dependency to remove warning

adrianclay committed:
Add TDD instructions to README
